ipatests/test_ipalib/test_rpc.py

```diff
@@ -341,3 +343,52 @@ def test_help_many_params(self):
                 "command 'system.methodHelp' takes at most 1 argument")
         else:
             raise AssertionError('did not raise')
+
+
+class test_rpcclient_context(PluginTester):
+    """
+    Test the context in `ipalib.rpc.rpcclient` plugin.
+    """
+    def setup(self):
+        try:
+            api.Backend.rpcclient.connect(ca_certfile='foo')
+        except (errors.NetworkError, IOError):
+            raise nose.SkipTest('%r: Server not available: %r' %
+                                (__name__, api.env.xmlrpc_uri))
+
+    def teardown(self):
+        if api.Backend.rpcclient.isconnected():
+            api.Backend.rpcclient.disconnect()
+
+    def test_context_cafile(self):
+        """
+        Test that ca_certfile is set in `ipalib.rpc.rpcclient.connect`
+        """
+        ca_certfile = getattr(context, 'ca_certfile', None)
+        assert_equal(ca_certfile, 'foo')
+
+    def test_context_principal(self):
+        """
+        Test that principal is set in `ipalib.rpc.rpcclient.connect`
+        """
+        principal = getattr(context, 'principal', None)
+        assert_equal(principal, 'admin@%s' % api.env.realm)
+
+    def test_context_request_url(self):
+        """
+        Test that request_url is set in `ipalib.rpc.rpcclient.connect`
+        """
+        request_url = getattr(context, 'request_url', None)
+        assert_equal(request_url, 'https://%s/ipa/session/json' % api.env.host)
+
+    def test_context_session_cookie(self):
+        """
+        Test that session_cookie is set in `ipalib.rpc.rpcclient.connect`
+        """
+        fuzzy_cookie = Fuzzy('^ipa_session=MagBearerToken=[A-Za-z0-9+\/+]=+;$')
+
+        session_cookie = getattr(context, 'session_cookie', None)
+        # pylint-2 is incorrectly spewing Too many positional arguments
+        # pylint: disable=E1121
+        unquoted = urllib.parse.unquote(session_cookie)
+        assert(unquoted == fuzzy_cookie)
```

Comment:
I believe `urllib.parser.unqoute` will return a `str` instance. By default, `Fuzzy` expects `unicode` type, you can either specify the type as in `Fuzzy(type=str)` or will need to do `assert(unicode(unquoted == fuzzy_cookie))`. The first approach would be preffered, though.

Reply:
It is a unicode on my system. Perhaps it maintains the current type.

Reply:
I did not test it, if it's unicode, it should be fine :)